edx_event_bus_redis/internal/consumer.py

```diff
@@ -0,0 +1,557 @@
+            DeserializingConsumer if it is.
+        """
+
+        consumer_config = load_common_settings()
+
+        # We do not deserialize the key because we don't need it for anything yet.
+        # Also see https://github.com/openedx/openedx-events/issues/86 for some challenges on determining key schema.
+        consumer_config.update({
+            'group.id': self.group_id,
+            'value.deserializer': None,
+            # Turn off auto commit. Auto commit will commit offsets for the entire batch of messages received,
+            # potentially resulting in data loss if some of those messages are not fully processed. See
+            # https://newrelic.com/blog/best-practices/redis-consumer-config-auto-commit-data-loss
+            'enable.auto.commit': False,
+        })
+
+        # create redis client and consumer.
+        return consumer_config
+
+    def _shut_down(self):
+        """
+        Test utility for shutting down the consumer loop.
+        """
+        self._shut_down_loop = True
+
+    def _consume_indefinitely(self):
+        """
+        Consume events from a topic in an infinite loop.
+        """
+
+        # This is already checked at the Command level, but it's possible this loop
+        # could get called some other way, so check it here too.
+        if not REDIS_CONSUMERS_ENABLED.is_enabled():
+            logger.error("Redis consumers not enabled, exiting.")
+            return
+
+        # .. setting_name: EVENT_BUS_REDIS_CONSUMER_CONSECUTIVE_ERRORS_LIMIT
+        # .. setting_default: None
+        # .. setting_description: If the consumer encounters this many consecutive errors, exit with an
+        #   error. This is intended to be used in a context where a management system (such as Kubernetes)
+        #   will relaunch the consumer automatically. The effect is that all runtime state is cleared,
+        #   allowing consumers in arbitrary "stuck" states to resume their work automatically. (The impetus
+        #   for this setting was a Django DB connection failure that was staying failed.) Process
+        #   managers like Kubernetes will use delays and backoffs, so this may also help with transient
+        #   issues such as networking problems or a burst of errors in a downstream service. Errors may
+        #   include failure to poll, failure to decode events, or errors returned by signal handlers.
+        #   This does not prevent committing of offsets back to the broker; any messages that caused an
+        #   error will still be marked as consumed, and may need to be replayed.
+        CONSECUTIVE_ERRORS_LIMIT = getattr(settings, 'EVENT_BUS_REDIS_CONSUMER_CONSECUTIVE_ERRORS_LIMIT', None)
+
+        try:
+            full_topic = get_full_topic(self.topic)
+            run_context = {
+                'full_topic': full_topic,
+                'consumer_group': self.group_id,
+                'expected_signal': self.signal,
+            }
+            # TODO: Consume events
```

Comment:
```suggestion

```

I think this is consuming events now?

Reply:
The actual work of fetching events from stream needs to be implemented. Actually the whole function would need to be updated to work with redis streams and consumer groups. I'll create separate tickets for producer and consumer implementation.